crypto/src/hash.rs

```diff
@@ -1053,7 +1097,7 @@ mod tests {
         .unwrap();
         let sig = Signature::from_base58_check(
```

Comment:
```suggestion
        let sig = Ed25519Signature::from_base58_check(
```

Why not use the more precise signature type directly?

Reply:
The base58 check is specifically for the `Signature` / `UnknownSignature` encoding - so this will be parsed as a `Signature::Unknown` variant, and then convert it.

I could change this to use `UnknownSignature` directly though?